app/ante/priority.go

```diff
@@ -0,0 +1,70 @@
+package ante
+
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	btcctypes "github.com/babylonlabs-io/babylon/x/btccheckpoint/types"
+	btclctypes "github.com/babylonlabs-io/babylon/x/btclightclient/types"
+	bstypes "github.com/babylonlabs-io/babylon/x/btcstaking/types"
+	bstctypes "github.com/babylonlabs-io/babylon/x/btcstkconsumer/types"
+	ftypes "github.com/babylonlabs-io/babylon/x/finality/types"
+)
+
+const (
+	// RegularTxMaxPriority is the max priority for transactions with regular messages.
+	// Thus, the reserved priority range for protocol liveness-related messages is (RegularTxMaxPriority, MaxInt64]
+	RegularTxMaxPriority = math.MaxInt64 - 1000
+	// LivenessTxPriority is the priority for protocol liveness-related messages.
+	// For the moment, the priority is the same for all of these messages
+	LivenessTxPriority = RegularTxMaxPriority + 100
```

Comment:
Do we want to add priority tiers?
'cause, for example, the `MsgCreateBTCDelegation` takes about [1.2M gas](https://testnet.babylon.explorers.guru/transaction/4A718959970785344789E03E81FAC0D178303610F9D0644EEF7FA5129101EE98?height=482328). We could set up a lower tier for the delegation creation to avoid censoring the other liveness-related txs that are cheaper (> 200k) when having many `MsgCreateBTCDelegation`

Reply:
we could have tiers for the ones that are really needed to the chain to work properly Top to Bottom priority:
- *btcctypes.MsgInsertBTCSpvProof,
- *btclighttypes.MsgInsertHeaders,
- *ftypes.MsgCommitPubRandList,
- *ftypes.MsgAddFinalitySig,

And maybe also add *ftypes.MsgSelectiveSlashingEvidence at the end 

The other types to create FPs/BTC delegations, add covenant signatures, undelegate can wait to be included